### PR TITLE
cherry-pick support channel last in BatchNorm*d(#27875)

### PR DESCRIPTION
**Pull request by iclementine**
### PR types

Bug fixes

### PR changes

OPs

### Describe

1. support channel last in BatchNorm*d (#27875)
2. fix a bug in batch_norm_op cuda kernel by extracting ResizeToChannelFist(Last), TransToChannelFirst(Last) to operators/layer_utils.h

### CODE REVIEW

**Review by jzhang533 — APPROVED** [review]
lgtm
